java-generic-mqtt/src/main/java/com/cumulocity/generic/mqtt/client/websocket/GenericMqttWebSocketFactory.java

```diff
@@ -0,0 +1,34 @@
+package com.cumulocity.generic.mqtt.client.websocket;
+
+import com.cumulocity.generic.mqtt.client.*;
+import com.cumulocity.generic.mqtt.client.model.GenericMqttMessage;
+import com.cumulocity.sdk.client.messaging.notifications.TokenApi;
+
+public class GenericMqttWebSocketFactory implements GenericMqttFactory {
+    private final String webSocketBaseUrl;
+    private final TokenApi tokenApi;
+
+    /**
+     * Constructs a GenericMqttWebSocketFactory instance and sets it to connect to the specified URI. The
+     * channel does not attempt to connect automatically. The connection will only be established once you
+     * obtain the instance of {@link GenericMqttPublisher} or {@link GenericMqttSubscriber} and invoke
+     * {@link GenericMqttPublisher#publish(GenericMqttMessage)} or {@link GenericMqttSubscriber#subscribe(GenericMqttMessageListener)}
+     *
+     * @param webSocketBaseUrl the server URI to connect to
+     * @param tokenApi         see {@link TokenApi}
```

Comment:
The @link to TokenApi should not be needed as the return type will be linked in docs. Maybe instead say something like '@param tokenApi the \<code\>TokenApi\</code> used to get \<code\>Token\</code\>s for the publishers and subscribers created by this factory.'

Reply:
Done